src/groupNormL2.jl

```diff
@@ -0,0 +1,81 @@
+# Group L2 norm (times a constant)
+
+export GroupNormL2
+
+"""
+**``L_2`` Group - norm**
+    GroupNormL2(λ = 1, g = 1, idx = [1, end])
+Returns the function
+```math
+f(x) =  \\sum\\_{i} \\lambda\\_{i}||x\\_{[i]}||\\_2)^{1/2}
+```
+for groups `x\\_{[i]}` a nonnegative weights `λ\\_i`.
+  Defaults to NormL2() in ProximalOperators if only 1 group is defined.
+"""
+struct GroupNormL2{
+  R <: Real,
+  RR <: AbstractVector{R},
+  I
+  } <: ProximableFunction
+  lambda::RR
+  idx::I
+
+  function GroupNormL2{R, RR,I}(
+    lambda::RR,
+    idx::I
+    ) where {R <: Real, RR <: AbstractVector{R}, I}
+    if any(lambda .< 0)
+      error("weights λ must be nonnegative")
+    elseif length(lambda) != length(idx)
+      error("number of weights and groups must be the same")
+    else
+      new{R, RR, I}(lambda, idx)
+    end
+  end
+end
+
+GroupNormL2(
+  lambda::RR = [1.],
+  idx::I = [Int[]],
```

Comment:
Here `idx` contains no indices. What does that mean? Intuitively, it seems like it means there is no ℓ2 penalty. But is that what you mean? Do you mean instead that *all* indices should be penalized? If so, does `idx = [1:end]` work?

Reply:
This is why I split it into cases; if length(lambda) = 1, we don't use idx at all, so it's ok if it isn't initialized. Your suggestion idx = [1:end] causes a compiler issue on my end.

Reply:
what's the error message?

Reply:
"syntax: missing last argument in "1:" range expression" because it thinks i'm ending the function.

Reply:
`idx:: I = [:]` appears to work.

Reply:
Wow should've thought of that.